components/dropdown/dropdown-opener-mixin.js

```diff
@@ -0,0 +1,132 @@
+export const DropdownOpenerMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			dropdownOpener: {
+				type: Boolean
+			},
+			noAutoOpen: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'no-auto-open'
+			},
+			disabled: {
+				type: Boolean,
+				reflect: true
+			}
+		};
+	}
+
+	constructor() {
+		super();
+		this.dropdownOpener = true;
+		this.noAutoOpen = false;
+		this.disabled = false;
+
+		this.__onKeyPress = this.__onKeyPress.bind(this);
+		this.__onMouseUp = this.__onMouseUp.bind(this);
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this.addEventListener('d2l-dropdown-open', this.__onOpened);
+		this.addEventListener('d2l-dropdown-close', this.__onClosed);
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+
+		requestAnimationFrame(() => {
+			const opener = this.getOpenerElement();
+			const content = this.__getContentElement();
+			if (!opener) {
+				return;
+			}
+			opener.setAttribute('aria-haspopup', 'true');
+			opener.addEventListener('keypress', this.__onKeyPress);
+			opener.addEventListener('mouseup', this.__onMouseUp);
+			opener.setAttribute('aria-expanded', (content && content.opened || false).toString());
+		});
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		const opener = this.getOpenerElement();
+		if (!opener) {
+			return;
+		}
+		opener.removeEventListener('keypress', this.__onKeyPress);
+		opener.removeEventListener('mouseup', this.__onMouseUp);
+	}
+
+	focus() {
+		const opener = this.getOpenerElement();
+		if (!opener) {
+			return;
+		}
+		opener.focus();
+	}
+
+	getOpener() {
+		return this;
+	}
+
+	getOpenerElement() {
+		return this;
+	}
+
+	toggleOpen(applyFocus) {
+		if (this.disabled) {
+			return;
+		}
+
+		const content = this.__getContentElement();
+		if (!content) {
+			return;
+		}
+		content.toggleOpen(applyFocus);
+	}
+
+	__getContentElement() {
+		return this.shadowRoot.querySelector('slot').assignedNodes().find(
+			node => node.nodeType === 1 && node.nodeName === 'D2L-DROPDOWN-CONTENT'
+		);
+	}
+
+	__onKeyPress(e) {
+		if (e.keyCode !== 13) {
+			return;
+		}
+		if (this.noAutoOpen) {
+			return;
+		}
+		this.toggleOpen(true);
+	}
+
+	__onMouseUp() {
+		if (this.noAutoOpen) {

```

Comment:
Super minor... if it were me I'd start to tidy up some of these to be one-liners. i.e..
```
if (this.noAutoOpen) return;
```
The original code was written at a time before we relaxed this practice a bit.  Don't feel like you have to, but if I were in here I probably would, simply just to make more code visible without costing readability.